R/loadFromMothur.R

```diff
@@ -82,22 +85,28 @@ loadFromMothur <- function(featureTableFile,
         stop("'sampleMetaFile' must be a single character value or NULL.",
             call. = FALSE)
     }
-    if(!.is_a_bool(featureNamesAsRefSeq)){
-        stop("'featureNamesAsRefSeq' must be TRUE or FALSE.", 
-            call. = FALSE)
-    }
-    if(!is.null(refSeqFile) && !.is_non_empty_string(refSeqFile)){
-        stop("'refSeqFile' must be a single character value or NULL.", 
-            call. = FALSE)
-    }
-    if(!is.null(phyTreeFile) && !.is_non_empty_string(phyTreeFile)){
-        stop("'phyTreeFile' must be a single character value or NULL.", 
-            call. = FALSE)
-    }
+  
+    # Tree and refseq files are not allowed
+    # if(!.is_a_bool(featureNamesAsRefSeq)){
+    #     stop("'featureNamesAsRefSeq' must be TRUE or FALSE.", 
+    #         call. = FALSE)
+    # }
+    # if(!is.null(refSeqFile) && !.is_non_empty_string(refSeqFile)){
+    #     stop("'refSeqFile' must be a single character value or NULL.", 
+    #         call. = FALSE)
+    # }
+    # if(!is.null(phyTreeFile) && !.is_non_empty_string(phyTreeFile)){
+    #     stop("'phyTreeFile' must be a single character value or NULL.", 
+    #         call. = FALSE)
+    # }
     #
     
     # Reads the featureTablefile 
-    feature_tab <- .read_mothur_feature(featureTableFile, ...)
+    feature_tab_and_data_to_colData <- .read_mothur_feature(featureTableFile, ...)
```

Comment:
As a discussion: The name `featureTableFile` is that used in the context of mothur? Is it a default name for mothur? I know, that the functions is informed a lot by `loadFromQIIME2`, but I think it not required to match the name.

I would put more emphasis on the view point of a typical mothur user. He has to recognize the file type by its name and know immediately, which file from mothur belong to which argument for `loadFromMothur`.

Is that the case for the argument names used here? I never have used mothur, so maybe someone else can offer an opinion? @antagomir @microsud

Reply:
I agree that it could be wise those change these names to match with Mothur

Reply:
E.g.

assay = shared
rowData = consTaxonomy
colData = design

However, I'm not sure if those are the names that are used in reality

Reply:
What do they use in their manual? If I remember correctly the `shared` name is also a documentation title, isn't it?

`taxonomyFile` and `designFile` are good names IMO. For the `sharedFile` a detailed entry on the man page might help as well.

Reply:
Yes, those names that I listed are the names that they use as documentation titles

Reply:
`constaxonomy` looks like a typo, but, hey, not our cup of coffee. `taxonomyFile` would probably match close enough, would it?

Reply:
It might, but they actually have two different taxonomy files:

- taxonomy file https://mothur.org/wiki/taxonomy_file/
- constaxonomy file https://mothur.org/wiki/constaxonomy_file/

Now only `constaxonomy `is allowed. Should `taxonomy `file be allowed aswell? I was thinking, if the name is `taxonomyFile,` then it could be misleading, because in reality it is `constaxonomy `file

Reply:
Good catch. I think you could extend the current function to support both.
 I would implement like this:
- read in file without header
- check whether the first lines contains valid columns names
- If yes proceed as `constaxonomy`
- If no:
  - check that two columns exist
  - process the first column as rowData column `sequenceName`
  - process the second column into taxonomic groups and check the result
  - check that the dimension of the taxonomy matched the assay

I think that should yield a valid result.